src/gateway/router/routes/registerContractRoute.ts

```diff
@@ -0,0 +1,175 @@
+import Router from '@koa/router';
+import { evalType } from '../../tasks/contractsMetadata';
+import { BUNDLR_NODE2_URL, BUNDLR_GATEWAY_URL } from '../../../constants';
+import { sleep } from 'warp-contracts';
+import { getCachedNetworkData } from '../../tasks/networkInfoCache';
+import { sendNotificationToCache } from '../../publisher';
+import { evalManifest } from './deployContractRoute';
+import { Tag } from 'arweave/node/lib/transaction';
+import { stringToB64Url } from 'arweave/node/lib/utils';
+import { fetch } from 'undici';
+import { backOff } from 'exponential-backoff';
+import { getTestnetTag } from './deployBundledRoute';
+
+const BUNDLR_QUERY = `query Transactions($ids: [String!]) {
+    transactions(ids: $ids) {
+      edges {
+        node {
+          address
+        }
+      }
+    }
+  }`;
+
+export async function registerContractRoute(ctx: Router.RouterContext) {
+  const { logger, gatewayDb } = ctx;
+
+  let initStateRaw = '';
+  let contractTx = null;
+  let txId = '';
```

Comment:
add bundlr node as another query param (and validate!)

Reply:
done